cg/services/deliver_files/delivery_file_tag_fetcher_service/bam_delivery_tags_fetcher.py

```diff
@@ -0,0 +1,28 @@
+from cg.constants import Workflow
+from cg.constants.housekeeper_tags import AlignmentFileTag
+from cg.services.deliver_files.delivery_file_tag_fetcher_service.delivery_file_tag_fetcher_service import (
+    FetchDeliveryFileTagsService,
+)
+from cg.services.deliver_files.delivery_file_tag_fetcher_service.error_handling import (
+    handle_tag_errors,
+)
+from cg.services.deliver_files.delivery_file_tag_fetcher_service.models import DeliveryFileTags
+
+
+class BamDeliveryTagsFetcher(FetchDeliveryFileTagsService):
+    """Class to fetch tags for bam files to deliver."""
+
+    @handle_tag_errors
+    def fetch_tags(self, workflow: Workflow) -> DeliveryFileTags:
+        """Fetch the tags for the bam files to deliver."""
+        return DeliveryFileTags(
+            case_tags=None,
+            sample_tags=[{AlignmentFileTag.BAM}],
+        )
+
+    @staticmethod
+    def _validate_workflow(workflow: Workflow):
+        """Validate the workflow."""
+        if workflow != Workflow.FASTQ:
+            raise ValueError(f"Workflow {workflow} is not supported for only BAM file delivery.")
+        return workflow
```

Comment:
I don't really see the need for this method. Also it seems to never be called

Reply:
added the call now. In cas someone tries to deliver bam files for something that is not RawData this should raise an error since that workflow might not contain any or doesn't have it on the sample level. So this is a user-error check